sdks/go/pkg/beam/core/runtime/exec/translate.go

```diff
@@ -64,6 +64,13 @@ func UnmarshalPlan(desc *fnpb.ProcessBundleDescriptor) (*Plan, error) {
 		}
 
 		u := &DataSource{UID: b.idgen.New()}
+		u.Coder, err = b.coders.Coder(cid) // Expected to be windowed coder
+		if err != nil {
+			return nil, err
+		}
+		if !coder.IsW(u.Coder) {
+			return nil, errors.Errorf("unwindowed coder %v on DataSource %v: %v", cid, id, u.Coder)
+		}
 
 		for key, pid := range transform.GetOutputs() {
```

Comment:
Do GRPC datasources always only have a single output?

I note that we aren't using the pid, in the new code.